l1-contracts/contracts/governance/PermanentRestriction.sol

```diff
@@ -183,4 +235,51 @@ contract PermanentRestriction is IRestriction, IPermanentRestriction, Ownable2St
             revert NotAnAdmin(admin, _potentialAdmin);
         }
     }
+
+    /// @notice Tries to get the new admin from the migration.
+    /// @param _call The call data.
+    /// @dev This function reverts if the provided call was not a migration call.
+    function tryGetNewAdminFromMigration(Call calldata _call) external view returns (address) {
+        if (_call.target != address(BRIDGE_HUB)) {
```

Comment:
Right now we use ChainAdmin contract as an admin in BridgeHub, SharedBridge, and other contracts. The logic is fine, just need to remember that ChainAdmin can be used only for Chains adminship

Reply:
I think we will need to deploy a different `chainadmin` contract for those ones

Reply:
(without the permanent rollup restriction)